code/backend/src/main/kotlin/ro/utcn/sd/icsaszar/project/controller/user/TeacherController.kt

```diff
@@ -0,0 +1,50 @@
+package ro.utcn.sd.icsaszar.project.controller.user
+
+import org.springframework.web.bind.annotation.*
+import ro.utcn.sd.icsaszar.project.dto.participation.ParticipationDTO
+import ro.utcn.sd.icsaszar.project.dto.participation.ParticipationStatusDTO
+import ro.utcn.sd.icsaszar.project.model.participation.ParticipationReviewStatus
+import ro.utcn.sd.icsaszar.project.model.report.TeacherReport
+import ro.utcn.sd.icsaszar.project.model.user.Teacher
+import ro.utcn.sd.icsaszar.project.service.report.ReportService
+import ro.utcn.sd.icsaszar.project.service.user.TeacherService
+
+@RestController
+@RequestMapping("/teacher")
+class TeacherController(
+        private val teacherService: TeacherService,
+        private val reportService: ReportService
+) {
+
+    @GetMapping("/participations")
+    fun getParticipations(@RequestParam(name = "status", required = false) status: ParticipationReviewStatus?): List<ParticipationDTO>{
+        val result = if(status == null)
+            teacherService.findAllParticipations()
+        else
+            teacherService.findParticipationsByStatus(status)
+        return result.map { it.toDTO() }
+    }
+
+    // This could also a PUT action (with the whole modified object as a body),
+    // and maybe it would be a better fit for this action,
+    // since all it is really doing is editing a field,
+    // but i feel like this solution has 2 advantages:
+    //    - less data is send (only the new status)
+    //    - it constrains to edit only to the relevant field
+    // Also, since teachers can only edit the status of participations,
+    // it won't cause confusion with adding new participations with a POST
+    // without parameters
+    @PostMapping("/participations")
```

Comment:
In this situation is fine, but you should generally try to avoid POST requests with query parameters. Even though it is OK from a HTTP point of view (https://stackoverflow.com/questions/611906/http-post-with-url-query-parameters-good-idea-or-not), generally it is a little misleading and can cause confusion for developers / consumers of your REST API, as typically it is not used.